build_tools/benchmarks/benchmark_helper.py

```diff
@@ -141,12 +146,61 @@ def _dump_cmds_handler(e2e_test_artifacts_dir: pathlib.Path,
   print(*lines, sep="\n")
 
 
+def merge_results(benchmark_results_files: List[io.TextIOWrapper]):
+  # Merges the benchmark results from `right` into `left` and returns the updated `left`
+  def _merge_two_resultsets(left, right):
+    # Parses a JSON benchmark results file and makes some sanity checks
+    def _load(file):
+      try:
+        content = json.load(file)
+      except json.JSONDecodeError as e:
+        raise RuntimeError('"{}" seems not to be a valid JSON file: {}'.format(
+            file.name, e.msg))
+      if not isinstance(content, dict):
+        raise RuntimeError(
+            '"{}" seems not to be a valid benchmark-results-file (No JSON struct as root element).'
+            .format(file.name))
+      if 'commit' not in content:
+        raise RuntimeError(
+            '"{}" seems not to be a valid benchmark-results-file ("commit" field not found).'
+            .format(file.name))
+      if 'benchmarks' not in content:
+        raise RuntimeError(
+            '"{}" seems not to be a valid benchmark-results-file ("benchmarks" field not found).'
+            .format(file.name))
+      return content
+
+    left_content = left if isinstance(left, dict) else _load(left)
+    right_content = right if isinstance(right, dict) else _load(right)
+
+    if left_content['commit'] != right_content['commit']:
+      raise RuntimeError(
+          '"{}" and the previous files are based on different commits ({} != {}). Merging not supported.'
+          .format(left.name, left_content['commit'], right_content['commit']))
+
+    left_content['benchmarks'].extend(right_content['benchmarks'])
+    return left_content
+
+  return functools.reduce(_merge_two_resultsets, benchmark_results_files)
+
+
+def _merge_results_handler(benchmark_results_files: List[io.TextIOWrapper],
```

Comment:
List can be a more generic Sequence
```suggestion
def _merge_results_handler(benchmark_results_files: Sequence[io.TextIOWrapper],
```

Reply:
Yes. Changed.